src/analyzer/protocol/smb/smb-pipe.pac

```diff
@@ -49,6 +49,22 @@ refine connection SMB_Conn += {
 
 		if ( it == fid_to_analyzer_map.end() )
 			{
+			// Too many analyzers?
+			if ( zeek::BifConst::SMB::max_dce_rpc_analyzers > 0 &&
+			     fid_to_analyzer_map.size() >= zeek::BifConst::SMB::max_dce_rpc_analyzers )
+				{
+				if ( smb_discarded_dce_rpc_analyzers )
+					zeek::BifEvent::enqueue_smb_discarded_dce_rpc_analyzers(zeek_analyzer(), zeek_analyzer()->Conn());
+
+				for ( const auto& kv : fid_to_analyzer_map )
+					{
+					kv.second->Done();
+					delete kv.second;
+					}
+
+				fid_to_analyzer_map.clear();
```

Comment:
I'm curious whether we need to clear out all of them or if we could either just drop the new one or some sort of FIFO queue. The actual SMB connection doesn't drop, so would we just start making more new DCE analyzers to go with it?

Reply:
The reproducer (and presumed behavior in the wild) is that the same pipe is opened and closed again over and over again, allocating a new `fid` and analyzer sequentially. In that scenario, dropping all analyzers or doing fifo style discard makes no difference.

Further, that code path only kicks in if we're missing SMB close requests due to capture loss / packet drops. If we see all close requests the analyzers map should mostly contain a single entry only.

If we discard a fid that is then re-used over the same connection, yes, it'll just re-instantiate a new one. Think that's okay as it shouldn't be the normal case.